src/register/pmpaddrx.rs

```diff
@@ -1,19 +1,24 @@
 use crate::register::pmpcfgx::Mode;
 use bit_field::BitField;
-use core::num::NonZeroUsize;
+use core::num::NonZeroU64;
 
 #[derive(Copy, Clone, Debug)]
 pub struct PmpAddr {
     bits: usize,
 }
 
+pub type Size = u64;
+pub type NonZeroSize = NonZeroU64;
+pub type Addr = u64;
+
 impl PmpAddr {
     #[inline]
-    pub fn decode(&self, mode: Mode) -> (usize, Option<NonZeroUsize>) {
+    pub fn decode(&self, mode: Mode) -> (Addr, Option<NonZeroSize>) {
+        let big_bits: Addr = self.bits as Addr;
         match mode {
-            Mode::OFF => (self.bits, None),
-            Mode::TOR => (self.bits << 2, None),
-            Mode::NA4 => (self.bits << 2, None),
+            Mode::OFF => (big_bits, None),
```

Comment:
Why are we returning an address in case of OFF? OFF means disabled and the address doesn't really mean anything. Probably should use Option<Addr> for this?

Reply:
Switched 'Option'.